sql-plugin/src/main/scala/com/nvidia/spark/rapids/PrioritySemaphore.scala

```diff
@@ -67,14 +69,15 @@ class PrioritySemaphore[T](val maxPermits: Int)(implicit ordering: Ordering[T])
       if (canAcquire(numPermits)) {
         commitAcquire(numPermits)
       } else {
-        waitingQueue.enqueue(info)
+        waitingQueue.add(info)
         while (!info.signaled) {
           info.condition.await()
         }
       }
     } catch {
-      case _: InterruptedException =>
-        info.interrupted = true
+      case e: Exception =>
+        waitingQueue.remove(info)
```

Comment:
Need to release the commits if info.signaled == true, because at that point we have taken ownership of our share of the semaphore but the exception will make the caller think we do not own it.